test/e2e-encryption/encryption_perf_test.go

```diff
@@ -0,0 +1,355 @@
+				//operatorencryption.AssertSecretsAndConfigMaps(t, clientSet, expectedMode, namespace, labelSelector)
+			},
+		})
+		return ts
+	}
+
+	library.TestEncryptionTypeIdentity(t, library.BasicScenario{
+		Namespace:                       operatorclient.GlobalMachineSpecifiedConfigNamespace,
+		LabelSelector:                   "encryption.apiserver.operator.openshift.io/component" + "=" + operatorclient.TargetNamespace,
+		EncryptionConfigSecretName:      fmt.Sprintf("encryption-config-%s", operatorclient.TargetNamespace),
+		EncryptionConfigSecretNamespace: operatorclient.GlobalMachineSpecifiedConfigNamespace,
+		OperatorNamespace:               operatorclient.OperatorNamespace,
+		TargetGRs:                       operatorencryption.DefaultTargetGRs,
+		AssertFunc: func(t testing.TB, clientSet library.ClientSet, expectedMode configv1.EncryptionType, namespace, labelSelector string) {
+			ts = time.Now()
+			// TODO: check how fast it is
+			//operatorencryption.AssertSecretsAndConfigMaps(t, clientSet, expectedMode, namespace, labelSelector)
+		},
+	})
+	return ts
+}
+
+func watchForMigrationControllerProgressingConditionAsync(t testing.TB, operatorClient operatorv1client.KubeAPIServerInterface, migrationStartedCh chan time.Time) {
+	t.Helper()
+	go watchForMigrationControllerProgressingCondition(t, operatorClient, migrationStartedCh)
+}
+
+func watchForMigrationControllerProgressingCondition(t testing.TB, operatorClient operatorv1client.KubeAPIServerInterface, migrationStartedCh chan time.Time) {
+	t.Helper()
+
+	waitPollInterval := 10 * time.Second
+	waitPollTimeout := 60 * time.Minute // a happy path scenario needs to roll out 3 revisions each taking ~10 min
+
+	t.Logf("Waiting up to %s for the condition %q with the reason %q to be set to true", waitPollTimeout.String(), "EncryptionMigrationControllerProgressing", "Migrating")
+	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		apiServerOperator, err := operatorClient.Get("cluster", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range apiServerOperator.Status.Conditions {
+			if cond.Type == "EncryptionMigrationControllerProgressing" && cond.Status == operatorv1.ConditionTrue {
+				t.Logf("EncryptionMigrationControllerProgressing condition observed at %v", cond.LastTransitionTime)
+				migrationStartedCh <- cond.LastTransitionTime.Time
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Logf("failed waiting for the condition %q with the reason %q to be set to true", "EncryptionMigrationControllerProgressing", "Migrating")
+	}
+}
+
+func populateDatabase(t testing.TB) {
+	t.Helper()
+	start := time.Now()
+	defer func() {
+		end := time.Now()
+		t.Logf("Populating etcd took %v", end.Sub(start))
+	}()
+
+	r := newRunner()
+	loaderFun := repeat(1, true,
+		createNamespace,
+		waitUntilNamespaceActive,
+		// TODO: consider adding ConfigMaps
+		repeatParallel(9010, 50, false, createSecret, reportSecret))
```

Comment:
this is probably too much given the current speed of migration.

Reply:
how much time do we permit for e2e?

Reply:
4h for `e2e-encryption` tests and 1h for `standard-e2e` tests.

Do we want to have a separate job just for the load tests?